File: mobile/lib/dtos/address_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'address_dto.g.dart';

@JsonSerializable()
class AddressDTO {
  String? id;
  String? recipientFirstName;
  String? recipientLastName;
  String? addressLine1;
  String? addressLine2;
  String? zipcode;
  String? city;
  String? state;

  AddressDTO({
    this.id,
    this.recipientFirstName,
    this.recipientLastName,
    this.addressLine1,
    this.addressLine2,
    this.zipcode,
    this.city,
    this.state,
  });

  factory AddressDTO.fromJson(Map<String, dynamic> json) =>
      _$AddressDTOFromJson(json);

  Map<String, dynamic> toJson() => _$AddressDTOToJson(this);
}

File: bff/src/main/kotlin/me/lucasbatista/vienna/sdk/dto/AddressDTO.kt
package me.lucasbatista.vienna.sdk.dto

data class AddressDTO(
    var id: String? = null,
    var recipientFirstName: String? = null,
    var recipientLastName: String? = null,
    var addressLine1: String? = null,
    var addressLine2: String? = null,
    var zipcode: String? = null,
    var city: String? = null,
    var state: String? = null,
    var country: String? = null,
)